src/main/java/com/uber/profiling/YamlConfigProvider.java

```diff
@@ -0,0 +1,186 @@
+                    }
+                }
+                
+                if (overrideMap != null) {
+                    for (Object key : overrideMap.keySet()) {
+                        String overrideKey = key.toString();
+                        Object valueObj = overrideMap.get(key);
+                        if (valueObj == null) {
+                            continue;
+                        }
+
+                        if (!(valueObj instanceof Map)) {
+                            logger.warn("Invalid override section: " + key + ": " + valueObj);
+                            continue;
+                        }
+                        
+                        Map<Object, Object> overrideValues = (Map<Object, Object>)valueObj;
+                        for (Map.Entry<Object, Object> entry : overrideValues.entrySet()) {
+                            if (entry.getValue() == null) {
+                                continue;
+                            }
+                            String configKey = entry.getKey().toString();
+                            addConfig(result, overrideKey, configKey, entry.getValue());
+                        }
+                    }
+                }
+
+                return result;
+            }
+        } catch (Throwable e) {
+            logger.warn("Failed to read config file: " + filePath, e);
+            return new HashMap<>();
+        }
+    }
+    
+    private void addConfig(Map<String, Map<String, List<String>>> config, String override, String key, Object value) {
+        Map<String, List<String>> configMap = config.computeIfAbsent(override, k -> new HashMap<>());
+        List<String> configValueList = configMap.computeIfAbsent(key, k -> new ArrayList<>());
+        
+        if (value instanceof List) {
+            for (Object entry : (List)value) {
+                configValueList.add(entry.toString());
+            }
+        } else if (value instanceof Object[]) {
+            for (Object entry : (Object[])value) {
+                configValueList.add(entry.toString());
+            }
+        } else {
+            configValueList.add(value.toString());
+        }
+    }
+
+    private byte[] getHttp(String url) {
+        try {
+            logger.debug(String.format("Getting url: %s", url));
+            try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+                HttpGet httpGet = new HttpGet(url);
+                try (CloseableHttpResponse httpResponse = httpClient.execute(httpGet)) {
+                    int statusCode = httpResponse.getStatusLine().getStatusCode();
+                    if (statusCode != HttpURLConnection.HTTP_OK) {
+                        logger.warn("Failed getting url: " + url + ", response code: " + statusCode);
+                        return new byte[0];
+                    }
+                    return IOUtils.toByteArray(httpResponse.getEntity().getContent());
```

Comment:
does this handle the http response charset properly?

Reply:
Good point. Will add a TODO